File: Assets/Scripts/GameManager.js
#pragma strict

//this class is designed to store all of the gameobjects into tables so that we can more easily access them.
//I thought of this when I was playing around with some RTS resource tutorials. It's built to be easily modifiable.
//Let me know if anyone has a better system.

var buildings:GameObject[];
//this has not been implement yet, but is a temp object that will point to a table of buildings. Will work once buildings
//are implemented.

var units:GameObject[];
//Thinking about this, we might want to separate units by their different types.
//we also don't have a way to divide teams yet to control who controls which units.
// we could do it somewhere in the game manager, or we could do it in the unit class itself.

function Start () {
	
}

function addToArray(addition:GameObject, array:GameObject[]):GameObject[]{
    var tempArray:GameObject[] = new GameObject[array.length+1];
    for (var i = 0; i < array.length; i++){
        tempArray[i] = array[i];
    }
    tempArray[tempArray.length-1] = addition;
    return tempArray;
}


function addBuildings(source:GameObject){
        buildings = addToArray(source, buildings);
}

    function addUnits(source:GameObject){
        units = addToArray(source, units);
    }

File: Assets/Scripts/GameManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;


public class GameManager : MonoBehaviour
	{


	public GameObject[] buildings;

	public GameObject[] units;


	public GameObject[] resources;

	public int money = 0;

	public UnitEnum unitToSpawn = UnitEnum.Building;
	public int unitCost = 0;



	public GameObject[] addToArray (GameObject addition, GameObject[] array){
		var tempArray = new GameObject[array.Length + 1];
		for (int i = 0; i< array.Length; i++){
			tempArray [i] = array [i];
		}
		tempArray [tempArray.Length - 1] = addition;
		return tempArray;
	}

	public void addBuildings(GameObject source){
		buildings = addToArray(source, buildings);
	}
	public void addUnits(GameObject source){
		units = addToArray(source, units);
	}
	public void addResources(GameObject source){
		resources = addToArray(source, resources);
	}

}

